plugins/modules/nd_backup_schedule.py

```diff
@@ -0,0 +1,210 @@
+    encryption_key: testtest2
+    frequency: 30
+    remote_location: test
+    state: present
+
+- name: Query a backup schedule
+  cisco.nd.nd_backup_schedule:
+    name: backupschedule1
+    state: query
+  register: query_one
+
+- name: Query all backup schedules
+  cisco.nd.nd_backup_schedule:
+    state: query
+  register: query_all
+
+- name: Delete a backup schedule
+  cisco.nd.nd_backup_schedule:
+    name: backupschedule1
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+
+import datetime
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.nd.plugins.module_utils.nd import NDModule, nd_argument_spec
+from ansible_collections.cisco.nd.plugins.module_utils.utils import snake_to_camel
+
+
+def main():
+    argument_spec = nd_argument_spec()
+    argument_spec.update(
+        name=dict(type="str"),
+        encryption_key=dict(type="str", no_log=True),
+        remote_location=dict(type="str"),
+        frequency=dict(type="int"),
+        scheduler_date=dict(type="str", aliases=["scheduler_start_date", "start_date", "date"]),
+        scheduler_time=dict(type="str", aliases=["scheduler_start_time", "start_time", "time"]),
+        backup_type=dict(type="str", default="config_only", choices=["config_only", "full"], aliases=["type"]),
+        state=dict(type="str", default="present", choices=["present", "query", "absent"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "backup", ["name", "encryption_key"]],
+            ["state", "absent", ["name"]],
+        ],
+    )
+
+    nd = NDModule(module)
+
+    name = nd.params.get("name")
+    encryption_key = nd.params.get("encryption_key")
+    remote_location = nd.params.get("remote_location")
+    frequency = nd.params.get("frequency")
+    scheduler_date = nd.params.get("scheduler_date")
+    scheduler_time = nd.params.get("scheduler_time")
+    backup_type = snake_to_camel(nd.params.get("backup_type"))
+    state = nd.params.get("state")
+
+    start_date_time = None
+    if scheduler_date and scheduler_time:
+        date_object = datetime.datetime.strptime(scheduler_date, "%Y-%m-%d")
+        time_object = datetime.datetime.strptime(scheduler_time, "%H:%M:%S")
+        start_date_time = "{:04d}-{:02d}-{:02d}T{:02d}:{:02d}:{:02d}Z".format(
+            date_object.year, date_object.month, date_object.day, time_object.hour, time_object.minute, time_object.second
+        )
+
+    path = "/api/v1/infra/backups/schedules"
+    schedules = nd.query_obj(path)
+    if name:
+        for schedule in schedules.get("schedules"):
+            if name == schedule.get("name"):
+                nd.previous = nd.existing = schedule
+                path = "{0}/{1}".format(path, name)
+                break
+    else:
+        nd.existing = schedules
+
+    if state == "present":
+        if nd.existing and nd.existing.get("name") == name:
```

Comment:
Can we simplify this as shown below if applicable?
```
payload = {
    "encryptionKey": encryption_key,
    "name": name,
    "type": backup_type,
    "frequency": frequency,
    "remoteLocation": remote_location,
    "startTime": start_date_time,
}

if nd.existing and nd.existing.get("name") == name:
    payload["frequency"] = frequency or nd.existing.get("frequency")
    payload["remoteLocation"] = remote_location or nd.existing.get("remoteLocation")
    payload["startTime"] = start_date_time or nd.existing.get("startTime")
```

Reply:
Done!